dcpy/models/base.py

```diff
@@ -131,3 +132,50 @@ def from_yaml(cls, yaml_str: str, *, template_vars=None) -> typing.Self:
     def from_path(cls, path: Path, *, template_vars=None) -> typing.Self:
         with open(path, "r", encoding="utf-8") as raw:
             return cls.from_yaml(raw.read(), template_vars=template_vars)
+
+
+class ModelWithDataFrame(BaseModel, arbitrary_types_allowed=True):
```

Comment:
@alexrichey - this is all for the purpose of testing and checking equality of objects like `comparison.Report`. I guess on one hand, with a pd DataFrame we're sort of getting outside of pydantic's wheelhouse (DFs are not super easily serializable - though I guess you could go to a json representation of the data), and I have no plans of dumping these to json or yml (or reading them). I still prefer these to dataclasses, mainly in that they serve an overlapping need and I'd rather stick with the object whose api I'm familiar with even if a dataclass might be more specific to my need here.

And I'd still have to write this equality function for a dataclass doing the same thing, since the same issue would exist.

Anyways just rambling and curious if you have any big thoughts

Reply:
Yeah, I agree. I think Pydantic classes avoid some pitfalls of [regular|data]classes and provide better documentation, so happy to just default to inheriting from BaseModel everywhere